servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/gc/AbstractRestGCTest.java

```diff
@@ -0,0 +1,416 @@
+    CommitOutput table2 =
+        commitSingleOp(
+            prefix, branch, branch.getHash(), 42, "cid_2", "table_2", "file1", null, null);
+    table2 =
+        commitSingleOp(
+            prefix, branch, table2.hash, 43, "cid_2", "table_2", "file2", table2.content, null);
+    fillExpectedContents(branch, 4, expectedExpired);
+
+    final Instant cutoffTime = Instant.now();
+
+    // commits for "table_1"
+    table1 =
+        commitSingleOp(
+            prefix, branch, table1.hash, 44, "cid_1", "table_1", "file3", table1.content, null);
+    table1 =
+        commitSingleOp(
+            prefix, branch, table1.hash, 45, "cid_1", "table_1", "file4", table1.content, null);
+    fillExpectedContents(branch, 2, expectedLive);
+    // commits for "table_2"
+    table2 =
+        commitSingleOp(
+            prefix, branch, table2.hash, 44, "cid_2", "table_2", "file3", table2.content, null);
+
+    fillExpectedContents(branch, 1, expectedLive);
+
+    // totally one live commit on table_2, two live commits on table_1.
+    // two expired commits on each table_1 and table_2.
+    performGc(
+        cutoffTime,
+        null,
+        expectedLive,
+        expectedExpired,
+        new HashSet<>(Arrays.asList(prefix + "_cid_1", prefix + "_cid_2")));
+  }
+
+  @Test
+  public void testSingleRefDropTable() throws BaseNessieClientServerException {
+    String prefix = "test2";
+    Map<String, Set<Content>> expectedExpired = new HashMap<>();
+    Map<String, Set<Content>> expectedLive = new HashMap<>();
+    Branch branch = createBranch("singleRefDropTable");
+    // two commits for "table_1"
+    CommitOutput table1 =
+        commitSingleOp(
+            prefix, branch, branch.getHash(), 42, "cid_1", "table_1", "file1", null, null);
+    table1 =
+        commitSingleOp(
+            prefix, branch, table1.hash, 43, "cid_1", "table_1", "file2", table1.content, null);
+    // two commits for "table_2"
+    CommitOutput table2 =
+        commitSingleOp(
+            prefix, branch, branch.getHash(), 42, "cid_2", "table_2", "file1", null, null);
+    table2 =
+        commitSingleOp(
+            prefix, branch, table2.hash, 43, "cid_2", "table_2", "file2", table2.content, null);
+    fillExpectedContents(branch, 4, expectedExpired);
+
+    final Instant cutoffTime = Instant.now();
+
+    // two commit for "table_1"
+    table1 =
+        commitSingleOp(
+            prefix, branch, table1.hash, 44, "cid_1", "table_1", "file3", table1.content, null);
+    table1 =
+        commitSingleOp(
+            prefix, branch, table1.hash, 45, "cid_1", "table_1", "file4", table1.content, null);
+    fillExpectedContents(branch, 2, expectedLive);
+
+    // commit for "table_2"
+    table2 =
+        commitSingleOp(
+            prefix, branch, table2.hash, 44, "cid_2", "table_2", "file3", table2.content, null);
+    // drop table "table_2"
+    dropTableCommit(prefix, branch, table2.hash, "table_2");
+    fillExpectedContents(branch, 2, expectedExpired);
```

Comment:
Why is this expired? It's committed after cut-off-time and table2.hash is still a live commit, right?

Reply:
dropping this table in line 119 and there is only one reference. so it is not live commit.